docs/development/best-practices.md

```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
 For `WARN`, `ERROR` and `FATAL` log messages use logger metadata.
 Also use logger metadata the result is a complex metadata object needing a multiple-line pretty stringification.
 
-For `INFO` log messages inline the metadata into the log message.
+For `INFO`, `DEBUG` and `TRACE` log messages inline the metadata into the log message.
 Also inline the metadata if the metadata object is complex.
```

Comment:
I do not understand this sentence. It sounds contradictory to `use logger metadata the result is a complex metadata object` for me.

Reply:
Yes I think it should say something like "It is permissible to not inline metadata if it's complex, but also think first if so much data is necessary to be logged".